platform/linux-generic/include/odp_packet_io_internal.h

```diff
@@ -33,7 +32,7 @@ extern "C" {
 
 #define PKTIO_MAX_QUEUES 64
 /* Forward declaration */
-typedef union pktio_entry_u pktio_entry_t;
+typedef struct pktio_entry odp_pktio_entry_t;
```

Comment:
this data structure belongs to one specific pktio implementation, so should not be public?

I understand that native drivers depend on this symbol because of pktio_ops interface is operating on this data structure.

So in order to make future native drivers independent of specific pktio implementation, instead of opening this internal data structure definition, instead we should refactory the pktio_ops interface definition.

And this task we can do later, after we have 1 or 2 native drivers in shape, what do you think?

Reply:
It is working right now. Why wait?